InaraResponder/InaraResponder.cs

```diff
@@ -534,58 +534,64 @@ private void handleShipTransferInitiatedEvent(ShipTransferInitiatedEvent @event)
 
         private void handleShipRenamedEvent(ShipRenamedEvent @event)
         {
-            Ship currentShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.shipid);
-            Dictionary<string, object> currentShipData = new Dictionary<string, object>()
-            {
-                { "shipType", currentShip.EDName },
-                { "shipGameID", currentShip.LocalId },
-                { "shipName", currentShip.name },
-                { "shipIdent", currentShip.ident },
-                { "shipRole", currentShip.Role.invariantName },
-                { "isHot", currentShip.hot },
+            var currentShipData = new Dictionary<string, object>()
+            {
+                { "shipType", @event.edModel },
+                { "shipGameID", @event.shipid },
+                { "shipName", @event.name },
+                { "shipIdent", @event.ident },
                 { "isCurrentShip", true }
             };
+            var currentShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.shipid);
+            if (currentShip?.EDName == @event.edModel)
+            {
+                currentShipData.Add("shipRole", (currentShip?.Role ?? Role.MultiPurpose).invariantName);
```

Comment:
We can remove the `?` from `currentShip`.

Reply:
No. The `GetShip` method can return null a null value if the local ship id is null or not found int he shipyard.

Reply:
Apologies for not explaining myself :(

What I meant is that on line 546 we already checked that `currentShip` is not null.
Do we need to keep the null check in the lines of the "then" block?

Reply:
That's mostly true but it assumes that `@event.edModel` isn't null (probably true - it's very unlikely that it would be null but I'm trying to program defensively).

Reply:
I see your point.
In general, I prefer to have a specific check for null in these cases and skip sections of code (or return).
That makes explicit the "safety" assumptions and simplifies reading the code.

Reply:
Fair enough. I've added pushed a change to include more explicit null checks.